File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/property/update/PropertyTypeUpdate.js
define([ "stjs", "as/dto/common/update/FieldUpdateValue", "as/dto/common/update/ListUpdateValue" ], function(stjs, FieldUpdateValue, ListUpdateValue) {
	var PropertyTypeUpdate = function() {
		this.label = new FieldUpdateValue();
		this.description = new FieldUpdateValue();
		this.schema = new FieldUpdateValue();
		this.transformation = new FieldUpdateValue();
		this.metaData = new ListUpdateValue();
	};
	stjs.extend(PropertyTypeUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.update.PropertyTypeUpdate';
		constructor.serialVersionUID = 1;
		prototype.typeId = null;
		prototype.label = null;
		prototype.description = null;
		prototype.schema = null;
		prototype.transformation = null;
		prototype.metaData = null;

		prototype.getObjectId = function() {
			return this.getTypeId();
		};
		prototype.getTypeId = function() {
			return this.typeId;
		};
		prototype.setTypeId = function(typeId) {
			this.typeId = typeId;
		};
		prototype.getLabel = function() {
			return this.label;
		};
		prototype.setLabel = function(label) {
			this.label.setValue(label);
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description.setValue(description);
		};
		prototype.getSchema = function() {
			return this.schema;
		};
		prototype.setSchema = function(schema) {
			this.schema.setValue(schema);
		};
		prototype.getTransformation = function() {
			return this.transformation;
		};
		prototype.setTransformation = function(transformation) {
			this.transformation.setValue(transformation);
		};
		prototype.getMetaData = function() {
			return this.metaData;
		};
		prototype.setMetaDataActions = function(actions) {
			this.metaData.setActions(actions);
		};
	}, {
		typeId : "IPropertyTypeId",
		metaData : "ListUpdateValue"
	});
	return PropertyTypeUpdate;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/update/PropertyTypeUpdate.java
/*
 * Copyright 2018 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.property.update;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.id.IPropertyTypeId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 *
 */
@JsonObject("as.dto.property.update.PropertyTypeUpdate")
public class PropertyTypeUpdate implements IUpdate, IObjectUpdate<IPropertyTypeId>
{
    private static final long serialVersionUID = 1L;
    
    @JsonProperty
    private IPropertyTypeId typeId;

    @JsonProperty
    private FieldUpdateValue<String> label = new FieldUpdateValue<String>();
    
    @JsonProperty
    private FieldUpdateValue<String> description = new FieldUpdateValue<String>();
    
    @JsonProperty
    private FieldUpdateValue<String> schema = new FieldUpdateValue<String>();
    
    @JsonProperty
    private FieldUpdateValue<String> transformation = new FieldUpdateValue<String>();

    @Override
    @JsonIgnore
    public IPropertyTypeId getObjectId()
    {
        return getTypeId();
    }

    @JsonIgnore
    public IPropertyTypeId getTypeId()
    {
        return typeId;
    }

    @JsonIgnore
    public void setTypeId(IPropertyTypeId typeId)
    {
        this.typeId = typeId;
    }

    @JsonIgnore
    public FieldUpdateValue<String> getLabel()
    {
        return label;
    }

    @JsonIgnore
    public void setLabel(String label)
    {
        this.label.setValue(label);
    }
    
    @JsonIgnore
    public FieldUpdateValue<String> getDescription()
    {
        return description;
    }
    
    @JsonIgnore
    public void setDescription(String description)
    {
        this.description.setValue(description);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getSchema()
    {
        return schema;
    }

    @JsonIgnore
    public void setSchema(String schema)
    {
        this.schema.setValue(schema);
    }
    
    @JsonIgnore
    public FieldUpdateValue<String> getTransformation()
    {
        return transformation;
    }

    @JsonIgnore
    public void setTransformation(String transformation)
    {
        this.transformation.setValue(transformation);
    }
    
}
